zerver/lib/templates.py

```diff
@@ -96,16 +94,14 @@ def render_markdown_path(
     Note that this assumes that any HTML in the Markdown file is
     trusted; it is intended to be used for documentation, not user
     data."""
-    if context is None:
-        context = {}
```

Comment:
Hmm, what's the benefit of removing this? It feels messier to need to remember to check if it's `None` each place we access it.

Is the idea that this is the `Jinja2` context, and we don't want to have that be valid if we're not doing such rendering?

I think that might make sense, but if so we probably need to split the parameters -- I `html_settings_links` is a Markdown processor option set via `set_relative_settings_links`.

We can fix this by splitting a `markdown_options` field from the `context` field, which is surely a good idea, but before someone goes too deep in looking at this, @andersk is this file going to be more or less rewritten with your effort to move our /help/ type pages off Python-Markdown?

Reply:
It looks like the idea is that `context=None` will mean we aren’t rendering Jinja at all, which is different from rendering Jinja with an empty context.

My markdown-it-py rewrite is still in progress but I’m fine with rebasing that, and having less Jinja to worry about will make it easier.

Reply:
@andersk is correct -- it's to make the "pure markdown" mode triggered by not passing in a context.  This is referenced in the commit message:
> Decide whether to run Jinja2 based on if a context is passed in at all.  This restores the behavior in the initial 0a99fa2fd669 where a call to `rendar_markdown_path` could be made to just render markdown, and not some other unmentioned and unrelated templating language as well.


> I think that might make sense, but if so we probably need to split the parameters -- I `html_settings_links` is a Markdown processor option set via `set_relative_settings_links`.

That makes this into a much bigger diff than I'm willing to do right now.  I need the ability to send custom emails without multiple unwanted layers of templating engine ruining everything -- and untangling the markdown parameters here is going to rapidly get me into the fact that we're re-entrantly calling `render_markdown_path` with different context sets and options each time, which are modifying global settings each time they get entered in a way that can't possibly result in correct behaviour, and is all under-tested.

Consider this a spot-fix and not a rototill.  Anything bigger is likely not worth the time, given the `markdown-it-py` rewrite.

Reply:
Makes sense, thanks for explaining!